### PR TITLE
Add asicdump tool to view asic state

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: asicdump/asicdump.cpp [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+struct CmdOptions
+{
+    bool shortNames;
+    bool disableColors;
+    bool skipAttributes;
+    bool followReferences;
+};
+
+CmdOptions g_cmdOptions;
+std::map<sai_object_id_t, const TableMap*> g_oid_map;
```
Comment (qiluo-msft) [inline asicdump/asicdump.cpp]: >std:: [](start = 0, length = 5)

'std::' can be removed.

File: asicdump/asicdump.cpp [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    return options;
+}
+
+size_t get_max_attr_len(const TableMap& map)
+{
+    SWSS_LOG_ENTER();
+
+    size_t max = 0;
+
+    for (const auto&field: map)
```
Comment (qiluo-msft) [inline asicdump/asicdump.cpp]: add a blank?

File: asicdump/asicdump.cpp (hunk 1 above)
Comment (kcudnik) [inline asicdump/asicdump.cpp]: i use std:: every time in cpp

File: asicdump/asicdump.cpp (hunk 2 above)
Comment (kcudnik) [inline asicdump/asicdump.cpp]: ??

File: asicdump/asicdump.cpp [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+        ss << field.second;
+
+        std::cout << ss.str() << std::endl;
+    }
+}
+
+int main(int argc, char ** argv)
+{
+    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_DEBUG);
```
Comment (kcudnik) [inline asicdump/asicdump.cpp]: i want log enter to be logged in main to know when new process started, i will add leter on other notice enter

File: Makefile.am [hunk 4]
```diff
@@ -1 +1 @@
-SUBDIRS = meta lib vslib syncd player
+SUBDIRS = meta lib vslib syncd player asicdump
```
Comment (lguohan) [inline Makefile.am]: mellanox has a sdkdump to dump all sdk state, to differentiate with sdkdump, can we call it saidump? for the player, can we call it saiplayer?

Comment (stcheng) [inline Makefile.am]: i'll change player to saiplayer

File: asicdump/asicdump.cpp (hunk 2 above)
Comment (qiluo-msft) [inline asicdump/asicdump.cpp]: auto &field

or 

    auto& field

**Comment by kcudnik** [comment]
since this is strictly combined with redis maybe sairedisdump and sairedisplayer ?

**Pull request merged**